files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,537 @@ class AaaCfg(object):
+                elif (('ipv4_default_gwaddr' not in cfgdb_entry) and ('ipv4_default_gwaddr' in appdb_entry)):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], appdb_entry['ipv4_default_gwaddr'], "DEL")
+                    del appdb_entry['ipv4_default_gwaddr']
+
+                if (('ipv6_default_gwaddr' in cfgdb_entry) and ('ipv6_default_gwaddr' not in appdb_entry)):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], cfgdb_entry['ipv6_default_gwaddr'], "ADD")
+                    appdb_entry['ipv6_default_gwaddr'] = cfgdb_entry['ipv6_default_gwaddr']
+                elif (('ipv6_default_gwaddr' in cfgdb_entry) and (cfgdb_entry['ipv6_default_gwaddr'] != appdb_entry['ipv6_default_gwaddr'])):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], appdb_entry['ipv6_default_gwaddr'], "DEL")
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], cfgdb_entry['ipv6_default_gwaddr'], "ADD")
+                    appdb_entry['ipv6_default_gwaddr'] = cfgdb_entry['ipv6_default_gwaddr']
+                elif (('ipv6_default_gwaddr' not in cfgdb_entry) and ('ipv6_default_gwaddr' in appdb_entry)):
+                    self.mgmt_intf_default_gwaddr_set(intf_keys[0], appdb_entry['ipv6_default_gwaddr'], "DEL")
+                    del appdb_entry['ipv6_default_gwaddr']
+
+                self.app_db.set_entry(APP_MGMT_INTF_TABLE, cur_appdb_key, appdb_entry)
+        else:
+            if op == "CREATE" or op == "UPDATE" :
+                self.mgmt_intf_ip_prefix_set(intf_keys[0], intf_keys[1], "ADD")
+                if (('gwaddr' in data) and ((appdb_entry == {}) or ('gwaddr' not in appdb_entry))):
+                    self.mgmt_intf_gwaddr_set(intf_keys[0], intf_keys[1], data['gwaddr'], "ADD")
+                    appdb_entry['gwaddr'] = data['gwaddr']
+                elif (('gwaddr' in data) and ('gwaddr' not in cfgdb_entry) and ('gwaddr' in appdb_entry)):
+                    self.mgmt_intf_gwaddr_set(intf_keys[0], intf_keys[1], data['gwaddr'],"DEL")
+                    del appdb_entry['gwaddr']
+                elif (('gwaddr' in data) and (data['gwaddr'] != appdb_entry['gwaddr'])):
+                    self.mgmt_intf_gwaddr_set(intf_keys[0], intf_keys[1], appdb_entry['gwaddr'],"DEL")
+                    self.mgmt_intf_gwaddr_set(intf_keys[0], intf_keys[1], data['gwaddr'], "ADD")
+                    appdb_entry['gwaddr'] = data['gwaddr']
+
+                if appdb_entry == {}:
+                    appdb_entry["NULL"] = "NULL"
+
+                self.app_db.set_entry(APP_MGMT_INTF_TABLE, cur_appdb_key, appdb_entry)
+
+            else:
+                self.mgmt_intf_ip_prefix_set(intf_keys[0], intf_keys[1], "DEL")
+                if ('gwaddr' in data):
+                    self.mgmt_intf_gwaddr_set(intf_keys[0], intf_keys[1], data['gwaddr'], "DEL")
+                self.app_db.delete_entry(APP_MGMT_INTF_TABLE, cur_appdb_key)
```

Comment:
I dont see the handling of` forced_mgmt_routes@`. Can you add that as well?

Reply:
127.0.0.1:6379[4]> hgetall  "MGMT_INTERFACE|eth0|FC00:2::32/64"
1) "forced_mgmt_routes@"
2) "10.3.145.98/31,10.3.145.8,100.127.20.16/28,10.3.149.170/31,40.122.216.24,13.91.48.226,10.3.145.14"
3) "gwaddr"
4) "fc00:2::1"

Reply:
Prince - having this forced_mgmt_routes under MGMT_INTERFACE table is confusing (e.g IPv6 addr in keys but holding IPv4 routes in forced_mgmt_route) ...it would better be present in separate table like MGMT_ROUTE.

Correct me if I'm wrong, I believe, we have added this forced_mgmt_route when mgmt VRF was not present in the past, after the introduction of mgmt VRF, I believe we should define the path for migration for the existing deployments to mgmt VRF instead of using 'default' table for forced mgmt routes.

Reply:
Added support for handling forced_mgmt_routes.